alica_engine/src/engine/AlicaContext.cpp

```diff
@@ -131,30 +131,53 @@ VariableSyncModule& AlicaContext::editSyncModule()
     return _engine->editResultStore();
 }
 
-YAML::Node AlicaContext::initConfig(const std::string& configPath, const std::string& agentName)
+YAML::Node AlicaContext::initConfig(const std::vector<std::string>& configPaths, const std::string& agentName)
 {
     YAML::Node node;
-    std::string configFile;
-    try {
-        configFile = essentials::FileSystem::combinePaths(configPath, agentName);
-        configFile = essentials::FileSystem::combinePaths(configFile, "Alica.yaml");
-        node = YAML::LoadFile(configFile);
-        return node;
-    } catch (YAML::BadFile& badFile) {
-        if (Logging::isInitialized()) {
-            Logging::logWarn(LOGNAME) << "Agent wise config file could not be used, file: " << configFile << ", error details- " << badFile.msg
-                                      << ", will try to use the global config file";
-        } else {
-            std::cerr << "[WARN]" << LOGNAME << "Agent wise config file could not be used, file:  " << configFile << ", error details - " << badFile.msg
-                      << ", will try to use the global config file" << std::endl;
+    std::string configFileSearchPath, configFilePath;
+    for (const auto& folder : configPaths) {
+        configFileSearchPath = essentials::FileSystem::combinePaths(folder, agentName);
+        if (essentials::FileSystem::findFile(configFileSearchPath, "Alica.yaml", configFilePath)) {
+            try {
+                node = YAML::LoadFile(configFilePath);
+                return node;
+            } catch (YAML::BadFile& badFile) {
+                if (Logging::isInitialized()) {
+                    Logging::logWarn(LOGNAME) << "Agent wise config file could not be parsed, file: " << configFilePath << ", error details- " << badFile.msg
+                                              << ", will continue the search in the next config directory";
+                } else {
+                    std::cerr << "[WARN] " << LOGNAME << " Agent wise config file could not be parsed, file:  " << configFilePath << ", error details - "
+                              << badFile.msg << ", will continue the search in the next config directory" << std::endl;
```

Comment:
IMO maybe logging component itself should output a generic msg if it is used pre-initialization rather than having custom error msgs in multiple places

Reply:
yes, I can take a look into implementing something like this in the logging component, because we have this issue in other places as well eg. some user code using the logger before it is initialised. I have observed this when other engine components are set before the logger is set